### PR TITLE
Support non-default mysql database ports

### PR DESCRIPTION
**Pull request by weiiwang01**
### Overview

Enable WordPress charm to link with a MySQL database operating on ports other than the default 3306.

### Rationale

Currently, the WordPress charm lacks the capability to connect to MySQL databases that run on ports other than the default.

### Juju Events Changes

None.

### Module Changes

Cleanup type hints for better clarity.

### Library Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -555,28 +559,28 @@ def _wp_is_installed(self):
...
 
         Returns:
-            Hostname of database running on port 3306. None if no endpoints are provided.
-            Note that WordPress will throw MySQL Error when supplying a port with the hostname
-            (i.e. host:port).
+            Hostname and port of database. None if no endpoints are provided.
 
         Raises:
-            WordPressBlockedStatusException: Provided endpoint contains port other than 3306.
+            RuntimeError: Provided endpoint contains port other than 3306.
```
Comment (cbartz) [inline src/charm.py:574]:
```suggestion
            RuntimeError: Provided endpoint has unknown format.
```

File: src/charm.py [hunk 2]
```diff
@@ -555,28 +559,28 @@ def _wp_is_installed(self):
...
-            return None
+            return None, None
         host_port = endpoint.split(":")
-        if len(host_port) == 2 and host_port[1] != "3306":
-            raise exceptions.WordPressBlockedStatusException(f"Invalid port {host_port[1]}")
-        # The endpoint might not contain port, we assume it to be 3306. If not, it will be caught
-        # by `_test_database_connectivity` function later on.
-        return host_port[0]
+        if len(host_port) == 1:
+            return host_port[0], 3306
```
Comment (arturo-seijas) [inline src/charm.py:580]: Bit picky but, can you define the port in a constant?

File: src/charm.py [hunk 3]
```diff
@@ -555,28 +559,28 @@ def _wp_is_installed(self):
...
         """Retrieve a single database endpoint.
 
         Args:
             endpoint: An endpoint of format host:port
 
         Returns:
-            Hostname of database running on port 3306. None if no endpoints are provided.
-            Note that WordPress will throw MySQL Error when supplying a port with the hostname
-            (i.e. host:port).
+            Hostname and port of database. None if no endpoints are provided.
```
Comment (cbartz) [inline src/charm.py:571]:
```suggestion
            Hostname and port of database. These are None if no endpoints are provided.
```

File: src/charm.py (hunk 2 above)
Comment (weiiwang01) [reply src/charm.py]: Updated, thanks!

File: src/charm.py (hunk 1 above)
Comment (weiiwang01) [reply src/charm.py]: Nice catch, updated both, thanks!

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by cbartz — APPROVED** [review]

**Pull request merged**